File: src/test/java/com/example/pages/IframesPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class IframesPage extends SeleniumBasePage {

    private final By iframeLocator = By.id("frame1");
    private final By frameText = By.id("frame-text");

    public IframesPage(WebDriver driver) {
        super(driver);
    }

    public String readTextInsideFrame() {
        switchToFrame(iframeLocator);
        String text = getText(frameText);
        switchToDefault();
        return text;
    }
}

File: python/pages/iframes_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class IframesPage(BasePage):

    FRAME = (By.ID, "frame1")
    TEXT = (By.ID, "frame-text")

    def read_frame_text(self):
        self.switch_to_frame(self.FRAME)
        text = self.get_text(self.TEXT)
        self.switch_to_default()
        return text
